extras/ws2812b/ws2812b.h

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file   ws2812b.h
+ * @brief  ESP8266 driver for WS2812B
+ *
+ * This is a simple bit-banging driver for WS2812B.
+ *
+ * It will work also for WS2812 and possibly others,
+ * with small alterations.
+ *
+ * The WS2812B protocol takes roughly 1µs per bit,
+ * thus ~24µs per "pixel", plus 50µs to indicate the
+ * end of data.
+ *
+ * @note
+ * The GPIO must be configured for output before trying
+ * to set colors!
+ *
+ * @attention
+ * Due to the precise timing required, interrupts are
+ * disabled until the data is sent.
+ *
+ * @author Ondřej Hruška, (c) 2016
+ *
+ * MIT License
+ */
+
+#ifndef WS2812B_DRV_H
+#define WS2812B_DRV_H
+
+#include <stdint.h>
+#include <stddef.h> // size_t
+
+#include "espressif/esp_common.h" // sdk_os_delay_us
+#include "esp/gpio.h"
+
+/**
+ * @brief Struct for easy manipulation of RGB colors.
+ *
+ * Set components in the xrgb.c.r (etc.) and you will get
+ * the hex in xrgb.num.
+ */
+typedef union {
+
+    /** Struct for ccess to individual color components */
+    struct __attribute__((packed)) {
+        uint8_t b;
+        uint8_t g;
+        uint8_t r;
+        uint8_t reserved; /**< padding field, could be used for brightness (?) */
```

Comment:
I don't think you need a padding field here, sizeof(xrgb_t) will be the size of the largest thing in the union (ie the uint32_t).